lib/process_settings/abstract_monitor.rb

```diff
@@ -11,8 +11,8 @@ class SettingsPathNotFound < StandardError; end
   OnChangeDeprecation = ActiveSupport::Deprecation.new('1.0', 'ProcessSettings::Monitor')
 
   class AbstractMonitor
-    @@full_context_cache = {}
 
+    attr_accessor :full_context_cache
```

Comment:
Now I remember this `@@full_context_cache` code! Although we generally like to avoid `@@` variables, I think it's fine here. No need to change it.

Reply:
At this point this is mirroring the caching style that we utilize over in the `contextual_logger`. I think it's cleaner and less likely to reach the size limit if it's confined to the instance. I would leave it like this unless we see any pressing need to pull the cache to persist across instances of the Monitor class.